src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/caseworker/MoreTimeRequestedCallbackHandler.java

```diff
@@ -0,0 +1,271 @@
+    private final MoreTimeRequestRule moreTimeRequestRule;
+    private final CaseDetailsConverter caseDetailsConverter;
+    private final NotificationService notificationService;
+    private final NotificationsProperties notificationsProperties;
+    private final GeneralLetterService generalLetterService;
+    private final UserService userService;
+    private final String generalLetterTemplateId;
+
+    private static final String PREVIEW_SENTENCE = "The response deadline will be %s .";
+    private static final String ERROR_MESSAGE =
+        "There was a technical problem. Nothing has been sent. You need to try again.";
+    private static final String STANDARD_DEADLINE_TEXT = "You’ve been given an extra 14 days to respond to the"
+            + " claim made against you by %s.\n"
+            + "\n"
+            + "You now need to respond to the claim before 4pm on %s.\n"
+            + "\n"
+            + "If you don’t respond, you could get a County Court Judgment (CCJ). This may make it harder to get "
+            + "credit, such as a mobile phone contract, credit card or mortgage.";
+    private static final String RESPONSE_DEADLINE_PREVIEW = "responseDeadlinePreview";
+
+    @Autowired
+    public MoreTimeRequestedCallbackHandler(
+            EventProducer eventProducer,
+            ResponseDeadlineCalculator responseDeadlineCalculator,
+            MoreTimeRequestRule moreTimeRequestRule,
+            CaseDetailsConverter caseDetailsConverter,
+            NotificationService notificationService,
+            NotificationsProperties notificationsProperties,
+            GeneralLetterService generalLetterService,
+            UserService userService,
+            @Value("${doc_assembly.generalLetterTemplateId}") String generalLetterTemplateId) {
+        this.eventProducer = eventProducer;
+        this.responseDeadlineCalculator = responseDeadlineCalculator;
+        this.moreTimeRequestRule = moreTimeRequestRule;
+        this.caseDetailsConverter = caseDetailsConverter;
+        this.generalLetterService = generalLetterService;
+        this.notificationService = notificationService;
+        this.notificationsProperties = notificationsProperties;
+        this.userService = userService;
+        this.generalLetterTemplateId = generalLetterTemplateId;
+    }
+
+    @Override
+    protected Map<CallbackType, Callback> callbacks() {
+        return ImmutableMap.of(
+            CallbackType.ABOUT_TO_START, this::requestMoreTimeViaCaseworker,
+            CallbackType.ABOUT_TO_SUBMIT, this::sendNotifications
+        );
+    }
+
+    @Override
+    public List<CaseEvent> handledEvents() {
+        return EVENTS;
+    }
+
+    @Override
+    public List<Role> getSupportedRoles() {
+        return ROLES;
+    }
+
+    public AboutToStartOrSubmitCallbackResponse requestMoreTimeViaCaseworker(CallbackParams callbackParams) {
+        CallbackRequest callbackRequest = callbackParams.getRequest();
```

Comment:
I suggest spinning up a new class to actually do the "handling"; let this class's responsibility be linking the callback events to the right place, and have another class that knows how to "request more time via caseworker". That new class can then maintain its own state and you won't need to pass objects around as parameters so much, and it makes refactoring much easier.

Reply:
This is a good idea, but since this is a ctsc priority we will add the refactoring of this (and its connection to the general letter service) to the tech debt backlog, if you can agree on that?